File: server/internal/infra/persistence/model/ai.go
package model

import "time"

type AIProvider struct {
	ID        int64     `gorm:"column:id;primaryKey"`
	Name      string    `gorm:"column:name;size:100;not null"`
	Type      string    `gorm:"column:type;size:20;not null"`
	APIURL    string    `gorm:"column:api_url;type:text;not null;default:''"`
	APIKey    string    `gorm:"column:api_key;type:text;not null;default:''"`
	IsActive  bool      `gorm:"column:is_active;not null;default:true"`
	CreatedAt time.Time `gorm:"column:created_at;autoCreateTime"`
	UpdatedAt time.Time `gorm:"column:updated_at;autoUpdateTime"`
}

func (AIProvider) TableName() string { return "ai_provider" }

type AIModel struct {
	ID         int64     `gorm:"column:id;primaryKey"`
	ProviderID int64     `gorm:"column:provider_id;not null"`
	Name       string    `gorm:"column:name;size:100;not null"`
	ModelID    string    `gorm:"column:model_id;size:200;not null"`
	IsActive   bool      `gorm:"column:is_active;not null;default:true"`
	CreatedAt  time.Time `gorm:"column:created_at;autoCreateTime"`
	UpdatedAt  time.Time `gorm:"column:updated_at;autoUpdateTime"`
}

func (AIModel) TableName() string { return "ai_model" }

type AITaskLog struct {
	ID            int64     `gorm:"column:id;primaryKey"`
	TaskType      string    `gorm:"column:task_type;size:40;not null"`
	ModelName     string    `gorm:"column:model_name;size:200;not null;default:''"`
	ProviderName  string    `gorm:"column:provider_name;size:100;not null;default:''"`
	Status        string    `gorm:"column:status;size:20;not null;default:'pending'"`
	InputText     string    `gorm:"column:input_text;type:text;not null;default:''"`
	OutputText    string    `gorm:"column:output_text;type:text;not null;default:''"`
	ErrorMessage  *string   `gorm:"column:error_message;type:text"`
	DurationMs    int       `gorm:"column:duration_ms;not null;default:0"`
	TriggerSource string    `gorm:"column:trigger_source;size:40;not null;default:'manual'"`
	CreatedAt     time.Time `gorm:"column:created_at;autoCreateTime"`
	UpdatedAt     time.Time `gorm:"column:updated_at;autoUpdateTime"`
}

func (AITaskLog) TableName() string { return "ai_task_log" }

File: admin/src/services/ai.ts
import { request, getAuthToken, API_BASE_URL } from './http'

// ── Types ──

export interface AIProvider {
  id: number
  name: string
  type: 'openai' | 'openrouter' | 'gemini'
  apiUrl: string
  isActive: boolean
  createdAt: string
  updatedAt: string
}

export interface AIModel {
  id: number
  providerId: number
  providerName?: string
  providerType?: string
  name: string
  modelId: string
  isActive: boolean
  createdAt: string
  updatedAt: string
}

export interface CreateAIProviderReq {
  name: string
  type: string
  apiUrl: string
  apiKey: string
  isActive?: boolean
}

export interface UpdateAIProviderReq {
  name?: string
  type?: string
  apiUrl?: string
  apiKey?: string
  isActive?: boolean
}

export interface CreateAIModelReq {
  providerId: number
  name: string
  modelId: string
  isActive?: boolean
}

export interface UpdateAIModelReq {
  providerId?: number
  name?: string
  modelId?: string
  isActive?: boolean
}

export interface ModerationResult {
  approved: boolean
  reason: string
  score: number
}

export interface TitleResult {
  title: string
  shortUrl: string
}

export interface RewriteResult {
  content: string
}

// ── TaskLog ──

export interface AITaskLog {
  id: number
  taskType: string
  modelName: string
  providerName: string
  status: string
  inputText?: string
  outputText?: string
  errorMessage?: string
  durationMs: number
  triggerSource: string
  createdAt: string
  updatedAt: string
}

export interface AITaskLogListResp {
  items: AITaskLog[]
  total: number
  page: number
  size: number
}

export interface AITaskLogListParams {
  page?: number
  pageSize?: number
  taskType?: string
  status?: string
  search?: string
}

// ── Provider CRUD ──

export function listAIProviders() {
  return request<AIProvider[]>('/admin/ai/providers')
}

export function createAIProvider(data: CreateAIProviderReq) {
  return request<AIProvider>('/admin/ai/providers', {
    method: 'POST',
    body: data,
  })
}

export function updateAIProvider(id: number, data: UpdateAIProviderReq) {
  return request<AIProvider>(`/admin/ai/providers/${id}`, {
    method: 'PUT',
    body: data,
  })
}

export function deleteAIProvider(id: number) {
  return request<null>(`/admin/ai/providers/${id}`, {
    method: 'DELETE',
  })
}

// ── Model CRUD ──

export function listAIModels() {
  return request<AIModel[]>('/admin/ai/models')
}

export function createAIModel(data: CreateAIModelReq) {
  return request<AIModel>('/admin/ai/models', {
    method: 'POST',
    body: data,
  })
}

export function updateAIModel(id: number, data: UpdateAIModelReq) {
  return request<AIModel>(`/admin/ai/models/${id}`, {
    method: 'PUT',
    body: data,
  })
}

export function deleteAIModel(id: number) {
  return request<null>(`/admin/ai/models/${id}`, {
    method: 'DELETE',
  })
}

// ── AI 功能 ──

export function moderateComment(content: string) {
  return request<ModerationResult>('/admin/ai/moderate-comment', {
    method: 'POST',
    body: { content },
  })
}

export function generateTitle(content: string) {
  return request<TitleResult>('/admin/ai/generate-title', {
    method: 'POST',
    body: { content },
  })
}

export function rewriteContent(content: string, instruction: string) {
  return request<RewriteResult>('/admin/ai/rewrite-content', {
    method: 'POST',
    body: { content, instruction },
  })
}

export async function generateSummaryStream(
  content: string,
  onChunk: (text: string) => void,
): Promise<void> {
  const token = getAuthToken()
  const headers: Record<string, string> = { 'Content-Type': 'application/json' }
  if (token) {
    headers['Authorization'] = `Bearer ${token}`
  }

  const resp = await fetch(`${API_BASE_URL}/admin/ai/generate-summary/stream`, {
    method: 'POST',
    headers,
    body: JSON.stringify({ content }),
  })

  if (!resp.ok) {
    throw new Error(`请求失败（${resp.status}）`)
  }

  const reader = resp.body?.getReader()
  if (!reader) throw new Error('无法读取响应流')

  const decoder = new TextDecoder()
  let buffer = ''

  while (true) {
    const { done, value } = await reader.read()
    if (done) break

    buffer += decoder.decode(value, { stream: true })
    const lines = buffer.split('\n')
    buffer = lines.pop() || ''

    for (const line of lines) {
      if (!line.startsWith('data: ')) continue
      const data = line.slice(6)
      if (data === '[DONE]') return
      try {
        const parsed = JSON.parse(data) as { content?: string; error?: string }
        if (parsed.error) throw new Error(parsed.error)
        if (parsed.content) onChunk(parsed.content)
      } catch (e) {
        if (e instanceof Error && e.message !== data) throw e
      }
    }
  }
}

export async function rewriteContentStream(
  content: string,
  instruction: string,
  onChunk: (text: string) => void,
): Promise<void> {
  const token = getAuthToken()
  const headers: Record<string, string> = { 'Content-Type': 'application/json' }
  if (token) {
    headers['Authorization'] = `Bearer ${token}`
  }

  const resp = await fetch(`${API_BASE_URL}/admin/ai/rewrite-content/stream`, {
    method: 'POST',
    headers,
    body: JSON.stringify({ content, instruction }),
  })

  if (!resp.ok) {
    throw new Error(`请求失败（${resp.status}）`)
  }

  const reader = resp.body?.getReader()
  if (!reader) throw new Error('无法读取响应流')

  const decoder = new TextDecoder()
  let buffer = ''

  while (true) {
    const { done, value } = await reader.read()
    if (done) break

    buffer += decoder.decode(value, { stream: true })
    const lines = buffer.split('\n')
    buffer = lines.pop() || ''

    for (const line of lines) {
      if (!line.startsWith('data: ')) continue
      const data = line.slice(6)
      if (data === '[DONE]') return
      try {
        const parsed = JSON.parse(data) as { content?: string; error?: string }
        if (parsed.error) throw new Error(parsed.error)
        if (parsed.content) onChunk(parsed.content)
      } catch (e) {
        if (e instanceof Error && e.message !== data) throw e
      }
    }
  }
}

// ── TaskLog API ──

export function listAITaskLogs(params: AITaskLogListParams) {
  const query = new URLSearchParams()
  if (params.page) query.set('page', String(params.page))
  if (params.pageSize) query.set('pageSize', String(params.pageSize))
  if (params.taskType) query.set('taskType', params.taskType)
  if (params.status) query.set('status', params.status)
  if (params.search) query.set('search', params.search)
  return request<AITaskLogListResp>(`/admin/ai/task-logs?${query.toString()}`)
}

export function getAITaskLog(id: number) {
  return request<AITaskLog>(`/admin/ai/task-logs/${id}`)
}
